app/admin/agent_generator.py

```diff
@@ -0,0 +1,514 @@
+
+    return schema, identified_skills
+
+
+async def fix_validation_errors(
+    schema: Dict[str, Any], schema_errors: List[str], agent_errors: List[str]
+) -> Dict[str, Any]:
+    """Attempt to fix validation errors.
+
+    Args:
+        schema: The original schema
+        schema_errors: Schema validation errors
+        agent_errors: Agent validation errors
+
+    Returns:
+        Fixed schema
+    """
+    fixed_schema = schema.copy()
+
+    # Fix required fields
+    required_fields = ["name", "purpose", "personality", "principles"]
+    for field in required_fields:
+        if field not in fixed_schema or not fixed_schema[field]:
+            fixed_schema[field] = f"Default {field.capitalize()}"
+
+    # Fix model and temperature
+    if "model" not in fixed_schema or fixed_schema["model"] not in ALLOWED_MODELS:
+        fixed_schema["model"] = "gpt-4.1-nano"
+    if "temperature" not in fixed_schema or not (0 <= fixed_schema["temperature"] <= 2):
+        fixed_schema["temperature"] = 0.7
+
+    # Fix agent-specific issues
+    if "Owner is required" in agent_errors:
+        fixed_schema["owner"] = "system"
+    if "Fee percentage too high" in agent_errors and "fee_percentage" in fixed_schema:
+        fixed_schema["fee_percentage"] = 100
+    if (
+        any("autonomous" in error for error in agent_errors)
+        and "autonomous" in fixed_schema
+    ):
+        fixed_schema.pop("autonomous")
+
+    return fixed_schema
+
+
+async def identify_skills(prompt: str, client: OpenAI) -> Dict[str, Any]:
+    """Identify relevant skills from the prompt.
+
+    Args:
+        prompt: The natural language prompt
+        client: OpenAI client
+
+    Returns:
+        Dict containing skill configurations
+    """
+    # First attempt: Use keyword matching for common capabilities
+    skills_config = keyword_match_skills(prompt)
+
+    # If no skills found or for more complex prompts, use OpenAI
+    if not skills_config:
+        skills_config = await openai_match_skills(prompt, client)
+
+    return skills_config
+
+
+def keyword_match_skills(prompt: str) -> Dict[str, Any]:
+    """Match skills using keyword matching.
+
+    Args:
+        prompt: The natural language prompt
+
+    Returns:
+        Dict containing skill configurations
+    """
+    skills_config = {}
+    prompt_lower = prompt.lower()
+
+    for keyword, skill_mapping in SKILL_MAPPING.items():
+        if keyword.lower() in prompt_lower:
+            for skill_name, states in skill_mapping.items():
+                if skill_name not in skills_config:
+                    skills_config[skill_name] = {"enabled": True, "states": {}}
+
+                for state in states:
+                    skills_config[skill_name]["states"][state] = "public"
+
+    # Apply proper skill configurations
+    return apply_skill_defaults(skills_config)
+
+
+def apply_skill_defaults(skills_config: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
In this scenario, the default skill strategy may no longer be effective. I believe we should not consider the default skills. I will also ask Marouen for his opinion.